phpBB/phpbb/filesystem/filesystem.php

```diff
@@ -613,11 +613,8 @@ protected function phpbb_is_writable($file)
 			}
 			else
 			{
-				$handle = @fopen($file, 'c');
-
-				if (is_resource($handle))
+				if (is_writable($file))
```

Comment:
The problem here is that `phpbb_is_writable()` is to be used either in case of Windows OS or if `is_writable()` doesn't exist, see https://github.com/phpbb/phpbb/blob/5a55891d060ecd4e9011146b68bc1366ac663d90/phpBB/phpbb/filesystem/filesystem.php#L262

So, in theory there's a case when `is_writable()` being used here while doesn't exist .

Reply:
I have some doubts about the function itself, anyway.